spark/src/test/scala/org/apache/spark/sql/delta/DeltaTypeWideningSuite.scala

```diff
@@ -89,6 +92,19 @@ trait DeltaTypeWideningTestMixin extends SharedSparkSession with DeltaDMLTestUti
 
   def addSingleFile[T: Encoder](values: Seq[T], dataType: DataType): Unit =
       append(values.toDF("a").select(col("a").cast(dataType)).repartition(1))
+
+  // Float/Double partition values are serialized as string leading to loss of
+  // precision. Also checkAnswer treats -0f and 0f as different without tolerance.
+  def checkAnswerWithTolerance(df: DataFrame, expected: DataFrame, dataType: DataType): Unit = {
+    if (Seq(FloatType, DoubleType).contains(dataType)) {
+      checkAggregatesWithTol(
+        df.sort("value"),
+        expected.sort("value").collect(),
+        absTol = 0.001)
+    } else {
+      checkAnswer(df, expected)
+    }
+  }
```

Comment:
I was hoping this would work to compare floats but it doesn't.. I haven't found an alternative yet besides rolling out my own comparison

Reply:
Rolled out my own in the end